services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDispatcher.java

```diff
@@ -157,7 +157,8 @@ public void dispatch(
           throw throwable;
         }
 
-        if (statusCode < HttpStatus.SC_INTERNAL_SERVER_ERROR) {
+        // Do not mark storage node fast for 429 status code
+        if (statusCode < HttpStatus.SC_INTERNAL_SERVER_ERROR && statusCode != HttpStatus.SC_TOO_MANY_REQUESTS) {
```

Comment:
Great catch! Could you please write a test for it?

Reply:
Added integ test for it